src/main/groovy/nebula/plugin/publishing/verification/VerifyPublicationTask.groovy

```diff
@@ -42,17 +48,15 @@ class VerifyPublicationTask extends DefaultTask {
         } as Set<ResolvedDependencyResult>
     }
 
-    Map<String, DefinedDependency> collectDefinedDependencies(Configuration parentConfiguration, Map<String, DefinedDependency> collector) {
-        parentConfiguration.extendsFrom.each {
-            collectDefinedDependencies(it, collector)
-        }
-        parentConfiguration.getDependencies().each {
-            if (it instanceof ExternalDependency) {
-                ExternalDependency dependency = (ExternalDependency) it
-                String preferredVersion = dependency.getVersionConstraint().preferredVersion
-                collector.put("${it.group}:${it.name}".toString(), new DefinedDependency(parentConfiguration.name, preferredVersion))
-            }
-        }
-        return collector
+    private Map<Project, ViolationsContainer> getViolations() {
+        PublishVerificationPlugin.VerificationViolationsCollectorHolderExtension extension = project.rootProject.extensions
+                .findByType(PublishVerificationPlugin.VerificationViolationsCollectorHolderExtension)
+        extension.collector
+    }
+
+    private List<Object> getDefinedDependencies() {
+        project.configurations.collect { configuration ->
```

Comment:
should add a test for

```dependencies {
  implementation 'g:a:1.0.0'
  api 'g2:a2:1.0.0'
}```

to make sure you don't need to check if configurations are resolvable